src/linear_algebra/pi.lean

```diff
@@ -240,13 +240,26 @@ variables [semiring R] {φ ψ : ι → Type*} [∀i, add_comm_monoid (φ i)] [
   [∀i, add_comm_monoid (ψ i)] [∀i, module R (ψ i)]
 
 /-- Combine a family of linear equivalences into a linear equivalence of `pi`-types. -/
-@[simps] def pi (e : Π i, φ i ≃ₗ[R] ψ i) : (Π i, φ i) ≃ₗ[R] (Π i, ψ i) :=
+@[simps apply] def Pi_congr_right (e : Π i, φ i ≃ₗ[R] ψ i) : (Π i, φ i) ≃ₗ[R] (Π i, ψ i) :=
 { to_fun := λ f i, e i (f i),
   inv_fun := λ f i, (e i).symm (f i),
-  map_add' := λ f g, by { ext, simp },
   map_smul' := λ c f, by { ext, simp },
   left_inv := λ f, by { ext, simp },
-  right_inv := λ f, by { ext, simp } }
+  .. add_equiv.Pi_congr_right (λ j, (e j).to_add_equiv) }
+
+@[simp]
+lemma Pi_congr_right_refl : Pi_congr_right (λ j, refl R (φ j)) = refl _ _ := rfl
+
+@[simp]
+lemma Pi_congr_right_symm (e : Π i, φ i ≃ₗ[R] ψ i) :
+  (Pi_congr_right e).symm = (Pi_congr_right $ λ i, (e i).symm) := rfl
+
+@[simp]
+lemma Pi_congr_right_trans {χ : ι → Type*}
+  [Π j, add_comm_monoid (χ j)] [Π j, module R (χ j)]
```

Comment:
The section is quite short - can you add `χ` up in the `variables` line? I can't suggest on github as the line is slightly out of the context